### PR TITLE
Display core parking indicator when hovering the mouse over the processor graphs

### PR DESCRIPTION
**Pull request by wnstngs**
## Purpose
Starting with Windows 7, the thread scheduler and processor power manager are able to park processors for power saving purposes.  Parking a processor makes it unavailable for thread scheduling, which contributes to potentially putting this processor into a deep idle state and thus reducing power consumption. Parked cores are marked as "Parked" in Task Manager, but not in System Informer.

This PR adds support for displaying core parking status in the tooltip when hovering over the processor graphs. Previously there was a proposal to add such a feature: https://github.com/winsiderss/systeminformer/issues/1083

## Proposed changes
1) Function that determines whether a core is parked. This uses the powerful CPU Sets mechanism available since Windows 10. Currently, every logical processor has a CPU Set and this looks like the most convenient and reliable way to determine if a specific core is parked.
2) Display "Parked" in the tooltip if the core is parked. https://imgur.com/a/Z1iDT8W

## TODO?
1) Revise the tooltip formatting code, as in the current form it easily becomes even more confusing when adding new functionality (`format` indexes change depending on different conditions). For example, at least use named variables for indexes of certain components of the tooltip (load percentages, relationships, time, and so on).
2) Consider using the CPU Sets mechanism in more places.

### CODE REVIEW

**Comment by wnstngs** [comment]
I have read the CLA Document and I hereby sign the CLA

File: SystemInformer/syssccpu.c
Comment (jxy-s) [inline SystemInformer/syssccpu.c:1984]: Please use native functions where possible:
`GetCurrentProcess` -> `NtCurrentProcess`
`GetSystemCpuSetInformation` -> `NtQuerySystemInformationEx` + `SystemCpuSetInformation`

File: SystemInformer/syssccpu.c
Comment (jxy-s) [inline SystemInformer/syssccpu.c:1098]: Tooltips can be fussy. Might need to try to cache this information. Might be an opportunity to eliminate an allocation. Agreed the formatting indexes could do some cleanup.

File: SystemInformer/syssccpu.c
Comment (jxy-s) [inline SystemInformer/syssccpu.c:1996]: `SYSTEM_CPU_SET_INFORMATION` is documented as variable-sized:
>This is a variable-sized structure designed for future expansion. When iterating over this structure, use the size field to determine the offset to the next structure.

File: SystemInformer/syssccpu.c
Comment (jxy-s) [inline SystemInformer/syssccpu.c:1947]: Current process is optional if we aren't interested in `AllocatedToTargetProcess`. Since we aren't interested in that field we should probably pass `NULL` instead?

**Review by jxy-s — COMMENTED** [review]
@wnstngs thanks for the contribution! I agree with showing the parked state in the tooltip. I have some feedback. Either I can merge this as and and go back to address it myself, or you can do so. Let me know how you'd like to handle it.

File: SystemInformer/syssccpu.c
Comment (wnstngs) [reply SystemInformer/syssccpu.c]: Fixed: https://github.com/winsiderss/systeminformer/pull/2040/commits/4458096161c84278c458cbac55f57d1f73b636c1

Now instead of `GetSystemCpuSetInformation` we use `NtQuerySystemInformationEx` which supports the `SystemCpuSetInformation` class.

File: SystemInformer/syssccpu.c
Comment (wnstngs) [reply SystemInformer/syssccpu.c]: Nice observation, thanks!
Fixed: https://github.com/winsiderss/systeminformer/pull/2040/commits/5e510ad5bd74836c340c79b0bddf949f550ad1e6

File: SystemInformer/syssccpu.c
Comment (wnstngs) [reply SystemInformer/syssccpu.c]: That's right. As of https://github.com/winsiderss/systeminformer/pull/2040/commits/4458096161c84278c458cbac55f57d1f73b636c1 we use `NtQuerySystemInformationEx` in which we don't pass a handle to the actual process (but we do pass a valid indirected handle as requested by the routine).

File: SystemInformer/syssccpu.c
Comment (wnstngs) [reply SystemInformer/syssccpu.c:1098]: Unlike the other parameters, the parking state of a set can be changed quite often by the processor power manager, so I'm not sure about caching. But I think it'd be worth at least adding a delay of half a second or a bit more.

**Review by jxy-s — APPROVED** [review]
Looks reasonable to me as it is, I will merge this when I have time - there is some cleanup I want to do after it gets merged, specifically around the juggling the formatting of the string mentioned in the PR.

**Review by dmex — APPROVED** [review]

**Comment by wnstngs** [comment]
@jxy-s Thank you!

**Pull request merged**
